src/Elders.Cronus/Discoveries/EventStoreDiscoveryBase.cs

```diff
@@ -14,7 +17,26 @@ protected override DiscoveryResult<IEventStore> DiscoverFromAssemblies(Discovery
         IEnumerable<DiscoveredModel> GetModels(DiscoveryContext context)
         {
             var loadedTypes = context.Assemblies.Find<IEventStoreIndex>();
+            yield return new DiscoveredModel(typeof(EventToAggregateRootId), typeof(EventToAggregateRootId), ServiceLifetime.Transient);
             yield return new DiscoveredModel(typeof(TypeContainer<IEventStoreIndex>), new TypeContainer<IEventStoreIndex>(loadedTypes));
         }
     }
+
+    public class InMemoryEventStoreDiscovery : EventStoreDiscoveryBase
+    {
+        protected override DiscoveryResult<IEventStore> DiscoverFromAssemblies(DiscoveryContext context)
+        {
+            var baseResult = base.DiscoverFromAssemblies(context);
+            var inMemoryModels = GetModels(context);
+            return new DiscoveryResult<IEventStore>(baseResult.Models.Concat(inMemoryModels));
+        }
+
+        IEnumerable<DiscoveredModel> GetModels(DiscoveryContext context)
+        {
+            yield return new DiscoveredModel(typeof(IEventStore), typeof(InMemoryEventStore), ServiceLifetime.Transient);
+            yield return new DiscoveredModel(typeof(InMemoryEventStoreStorage), typeof(InMemoryEventStoreStorage), ServiceLifetime.Singleton);
+            yield return new DiscoveredModel(typeof(IIndexStore), typeof(InMemoryIndexStore), ServiceLifetime.Singleton);
+            yield return new DiscoveredModel(typeof(IEventStorePlayer), typeof(InMemoryEventStorePlayer), ServiceLifetime.Singleton);
```

Comment:
We need to verify if all these interfaces are overridden by the persistence and projections packages. There is a boolean flag .CanOverride which has to be set to true